slides/12_Remix.html

```diff
@@ -24,6 +24,727 @@
+        </section>
+        <section>
+          <h2>Exercise</h2>
+          <ul>
+            <li class="fragment">
+              Load the data for the list and detail pages from the Poke API
+            </li>
+            <li class="fragment">
+              Show the Not Found page if an unknown pokemon name is passed in
+              the URL
+            </li>
+          </ul>
+        </section>
+        <section>
+          <section>
+            <h2>Solution</h2>
+          </section>
+          <section>
+            <p><span class="code">fetcher.ts</span></p>
+            <pre><code class="hljs tsx" data-trim data-line-numbers="|4-6" data-id="code"><script type="text/template">
+              export async function fetcher<T>(uri: string, init?: RequestInit): Promise<T> {
+                const response = await fetch(uri, init);
+              
+                if (response.status === 404) {
+                  throw new Response(null, { status: 404, statusText: "Not found" });
+                }
+
+                if (!response.ok) throw new Error("Could not fetch data!");
+                return response.json();
+              }
+            </script></code></pre>
+          </section>
+          <section>
+            <p><span class="code">pokemon._index.tsx</span></p>
+            <pre><code class="hljs tsx" data-trim data-line-numbers="1-7|10|" data-id="code"><script type="text/template">
+              export async function loader() {
+                const pokemonList = await fetcher<PokemonResultDto>(
+                  "https://pokeapi.co/api/v2/pokemon?limit=1000",
+                );
+              
+                return json(pokemonList);
+              }
+              
+              export default function ListPage() {
+                const pokemonList = useLoaderData<typeof loader>();
+
+                return <PokeList pokemons={pokemonList.results} />;
+              }
+            </script></code></pre>
+          </section>
+          <section>
+            <p><span class="code">pokemon.$pokemonName.tsx</span></p>
+            <pre><code class="hljs tsx" data-trim data-line-numbers="1-7|10|" data-id="code"><script type="text/template">
+              export async function loader({ params }: LoaderArgs) {
+                const pokemonDetail = await fetcher<PokemonDetailDto>(
+                  `https://pokeapi.co/api/v2/pokemon/${params.pokemonName}`,
+                );
+              
+                return json(pokemonDetail);
+              }
+              
+              export default function DetailPage() {
+                const pokemonDetail = useLoaderData<typeof loader>();
+                
+                return (
+                  <div>
+                    <span>{pokemonDetail.name}</span>
+                    <img src={pokemonDetail.sprites.front_shiny} alt={pokemonDetail.name} />
+                  </div>
+                );
+              }
+            </script></code></pre>
```

Comment:
**Suggestion:** Add notes to analyze the exercise result also in the network tab to see what's going on in all the steps: Redirect, First page load, subsequent navigation load. I think it's important to highlight what magic happens under the hood :)

Reply:
Oh yeah, good idea!

Reply:
I added this to the exercise slide now. So the participants can use the time during the exercise to play with this.